examples/api/lib/material/autocomplete/autocomplete.4.dart

```diff
@@ -71,14 +71,11 @@ class _AsyncAutocompleteState extends State<_AsyncAutocomplete > {
     late final Iterable<String> options;
     try {
       options = await _FakeAPI.search(_currentQuery!, _networkEnabled);
-    } catch (error) {
-      if (error is _NetworkException) {
-        setState(() {
-          _networkError = true;
-        });
-        return <String>[];
-      }
-      rethrow;
+    } on _NetworkException {
+      setState(() {
```

Comment:
Since this uses setState after an await, consider doing:

```dart
} on _NetworkException {
  if (mounted) {
    setState(() {
      _networkError = true;
    });
  }

  return <String>[];
}
```

There is a proposal to catch this kind of thing with a lint, but the lint does not yet exist.

Reply:
Ooh, good call, thanks!